### PR TITLE
Filter constant strings from select clauses in Django queryset extra

### LINKED ISSUE #575: Ignore cases where constant strings are provided to django queryset `extra`
Example failing test cases: https://semgrep.live/gxJK

Observed while scanning https://github.com/DefectDojo/django-DefectDojo/blob/de98f281165dedffb27a0a394fb0bbb66ed0e72e/dojo/metrics/views.py#L57-L69

Having dataflow here would help a lot.

### PR DESCRIPTION
**Pull request by minusworld**
Fixes https://github.com/returntocorp/semgrep-rules/issues/575

### CODE REVIEW

File: python/django/security/audit/query-set-extra.yaml [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
-  severity: ERROR
-  pattern-either:
-  - pattern: $X.objects.$FUNC(...).extra(...)
-  - pattern: $X.objects.$FUNC(...).$FILTER(...).extra(...)
-  - pattern: $X.objects.$FUNC(...).$FILTER(...).$UPDATE(...).extra(...)
+  languages:
+  - python
+  severity: WARNING
+  patterns:
+  - pattern: $MODEL.extra(...)
```
Comment (inkz) [inline python/django/security/audit/query-set-extra.yaml:23]: maybe add smth like:
```
  - metavariable-regex:
      metavariable: $MODEL
      regex: .*objects.*
```
to filter non `QuerySet`s like (e.g. `foobar.extra('yo')`)

Comment (minusworld) [reply python/django/security/audit/query-set-extra.yaml:23]: I think that you can use `.extra` on querysets without the `objects` property. See https://docs.djangoproject.com/en/3.2/ref/models/querysets/#extra

I guess I should check to see how common `.extra` is and if it's prone to FPs.

**Review by colleend — APPROVED** [review]
lgtm, @inkz is this ok with you?

File: python/django/security/audit/query-set-extra.yaml (hunk 1 above)
Comment (inkz) [reply python/django/security/audit/query-set-extra.yaml:23]: oh ok, I get it now :+1:

**Comment by inkz** [comment]
> lgtm, @inkz is this ok with you?

yes!

**Pull request merged**